parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeospatialStatistics.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.statistics.geometry;
+
+import org.apache.parquet.Preconditions;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+
+/**
+ * A structure for capturing metadata for estimating the unencoded,
+ * uncompressed size of geospatial data written.
+ */
+public class GeospatialStatistics {
+
+  public static final String DEFAULT_GEOSPATIAL_STAT_CRS = "OGC:CRS84";
+
+  // Metadata that may impact the statistics calculation
+  private final String crs;
+  private final BoundingBox boundingBox;
+  private final EdgeInterpolationAlgorithm edgeAlgorithm;
+  private final GeospatialTypes geospatialTypes;
+  //   private final WKBReader reader = new WKBReader();
+
+  /**
+   * Whether the statistics has valid value.
+   *
+   * It is true by default. Only set to false while it fails to merge statistics.
+   */
+  private boolean valid = true;
+
+  public void mergeStatistics(GeospatialStatistics other) {
+    if (other == null) {
+      return;
+    }
+    this.boundingBox.merge(other.boundingBox);
+    this.geospatialTypes.merge(other.geospatialTypes);
```

Comment:
`this.boundingBox` and `this.geospatialTypes` can be null. There's a `NoopBuilder` creating such GeospatialStatistics objects for non-geospatial columns.